lib/pbench/test/functional/server/test_put.py

```diff
@@ -386,9 +387,33 @@ def test_list_filter_and_or(self, server_client: PbenchServerClient, login_user)
             ), f"Dataset {dataset.name} date is {date}"
 
 
+class TestUpdate:
+    @pytest.mark.dependency(name="publish", depends=["index"], scope="session")
+    @pytest.mark.parametrize("access", ("public", "private"))
+    def test_publish(self, server_client: PbenchServerClient, login_user, access):
```

Comment:
Is `depends=["index"]` sufficient?  We don't have any other tests which could be confused if they run after this test plays with the `access` values?

Reply:
The "index" test is what checks expected metadata after upload; nothing else cares. (But of course the delete test has to go last...)